backend/resources/transactions.py

```diff
@@ -44,11 +50,11 @@ def post(self):
 
 class TransactionsUser(Resource):
     @auth.login_required()
-    def get(self, id_user):
-        user = UsersModel.find_by_id(id_user)
+    def get(self, email):
+        user = UsersModel.find_by_email(email)
+        if g.user != user:
```

Comment:
Esta be l'ordre a diferencia de l'anterior, perque en cas de que el usuari sigui None la igualtat sera falsa. Pero crec que estaria millor l'ordre al reves, per notificar que l'usuari no existeix abans que un problema de autoritzacio.